File: app/src/main/java/com/estebanlamas/planesradar/presentation/map/di/MapModule.java
package com.estebanlamas.planesradar.presentation.map.di;

import android.support.v4.app.FragmentActivity;

import com.estebanlamas.planesradar.data.AircraftDataRepository;
import com.estebanlamas.planesradar.domain.GetAircraftsUseCase;
import com.estebanlamas.planesradar.domain.repository.AircraftRepository;
import com.estebanlamas.planesradar.presentation.map.MapPresenter;

import dagger.Module;
import dagger.Provides;

@Module
public class MapModule {
    private FragmentActivity activity;

    public MapModule(FragmentActivity activity) {
        this.activity = activity;
    }

    @Provides
    FragmentActivity provideActivity(){
        return this.activity;
    }

    @Provides
    public MapPresenter provideMapPresenter(GetAircraftsUseCase getAircraftsUseCase) {
        return new MapPresenter(getAircraftsUseCase);
    }

    @Provides
    public AircraftRepository provideAircraftRepository(AircraftDataRepository aircraftDataRepository) {
        return aircraftDataRepository;
    }

}

File: app/src/main/java/com/estebanlamas/planesradar/presentation/map/di/MapModule.kt
package com.estebanlamas.planesradar.presentation.map.di

import androidx.fragment.app.FragmentActivity

import com.estebanlamas.planesradar.data.AircraftDataRepository
import com.estebanlamas.planesradar.domain.GetAircraftsUseCase
import com.estebanlamas.planesradar.domain.repository.AircraftRepository
import com.estebanlamas.planesradar.presentation.map.MapViewModel

import dagger.Module
import dagger.Provides

@Module
class MapModule(private val activity: FragmentActivity) {

    @Provides
    fun provideActivity(): FragmentActivity {
        return this.activity
    }

    @Provides
    fun provideMapPresenter(getAircraftsUseCase: GetAircraftsUseCase): MapViewModel {
        return MapViewModel(getAircraftsUseCase)
    }

    @Provides
    fun provideAircraftRepository(aircraftDataRepository: AircraftDataRepository): AircraftRepository {
        return aircraftDataRepository
    }

}
